From 6c95b29117469a5ecb22655fb561dceee04106e8 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Zbigniew=20J=C4=99drzejewski-Szmek?= <zbyszek@in.waw.pl>
Date: Fri, 17 Jul 2020 07:44:10 +0200
Subject: [PATCH] hwdb: allow spaces in usb: matches and similar patterns

In the past we didn't have any matches like that, so the parser was stricter
than necessary, but now we have, so allow that.

(cherry picked from commit 457763aa0393a9f11e30071bb3794caf1a4f2f64)

Related: RHEL-130979
---
 hwdb/parse_hwdb.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/hwdb/parse_hwdb.py b/hwdb/parse_hwdb.py
index d15ae0470c..db26a5e4d4 100755
--- a/hwdb/parse_hwdb.py
+++ b/hwdb/parse_hwdb.py
@@ -89,7 +89,7 @@ def hwdb_grammar():
                 for category, conn in TYPES.items())
 
     matchline_typed = Combine(prefix + Word(printables + ' ' + '®'))
-    matchline_general = Combine(Or(GENERAL_MATCHES) + ':' + Word(printables))
+    matchline_general = Combine(Or(GENERAL_MATCHES) + ':' + Word(printables + ' ' + '®'))
     matchline = (matchline_typed | matchline_general) + EOL
 
     propertyline = (White(' ', exact=1).suppress() +
